soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -5229,6 +5274,70 @@ CustomMessage Randomizer::GetWarpSongMessage(u16 textId, bool mysterious) {
     return messageEntry;
 }
 
+CustomMessage Randomizer::GetMiscMessage(s16 scene, u16 originalTextId) {
```

Comment:
I don't see much benefit to this being one function that splits into 3 blocks vs just 3 different functions personally. And the scene argument only seems relevant for the middle block

That being said, I think a lot of these custom message handlers could be cleaned up so up to you if you want to make the switch, I think this system will look a bit different for v3/v4 :tm:

Reply:
i also think splitting it into 3 makes sense
`GetFrogMessage`
`GetSheikMessage`
`GetSariaMessage`

we're already making calls for each from separate places in `CustomMessage_RetrieveIfExists`, so we might as well call into different methods.

it could be cool to see a bunch of the logic in `CustomMessage_RetrieveIfExists` moved out and cleaned up, but that feels like it should be its own PR